src/ztgamemgr.rs

```diff
@@ -47,102 +48,43 @@ struct SYSTEMTIME {
     w_milliseconds: u16,
 }
 
-impl ZTGameMgr {
-    // returns the address of the ZTGameMgr instance in memory
-    fn instance() -> u32 {
-        get_from_memory::<u32>(GLOBAL_ZTGAMEMGR_ADDRESS)
-    }
-
-    // returns the SYSTEMTIME struct in memory
-    fn get_date() -> SYSTEMTIME {
-        let date = get_from_memory::<SYSTEMTIME>(Self::instance() + 0x1194);
-        date
-    }
-
-    // returns the cash in the player's account
-    fn get_cash() -> f32 {
-        get_from_memory::<f32>(Self::instance() + 0x0C)
-    }
-
-    // sets the cash in the player's account
-    fn set_cash(cash: f32) {
-        let cash_address = Self::instance() + 0x0C;
-        unsafe {
-            *(cash_address as *mut f32) = cash;
-        }
-    }
-
-    // adds cash to the player's account
-    fn add_cash(cash: f32) {
-        let current_cash = Self::get_cash();
-        Self::set_cash(current_cash + cash);
-    }
-
-    // subtracts cash from the player's account
-    fn sub_cash(cash: f32) {
-        let current_cash = Self::get_cash();
-        Self::set_cash(current_cash - cash);
-    }
-
-    // returns the number of animals in the zoo
-    fn num_animals() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x30)
-    }
-
-    // returns the number of species in the zoo
-    fn num_species() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x38)
-    }
-
-    // returns the number of guests in the zoo
-    fn num_guests() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x54)
-    }
-
-    // returns the number of tired guests in the zoo
-    fn num_tired_guests() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x3C)
-    }
-
-    // returns the number of hungry guests in the zoo
-    fn num_hungry_guests() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x40)
-    }
-
-    // returns the number of thirsty guests in the zoo
-    fn num_thirst_guests() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x44)
-    }
-
-    // returns the number of guests that need to use the restroom
-    fn num_guests_restroom_need() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x48)
-    }
-
-    // returns the number of guests in the filter
-    fn num_guests_in_filter() -> u32 {
-        get_from_memory::<u32>(Self::instance() + 0x54)
-    }
-
-    // returns the zoo admission cost
-    fn zoo_admission_cost() -> f32 {
-        get_from_memory::<f32>(Self::instance() + 0x1160)
-    }
+struct GameMgr {
+    enable_dev_mode: fn(bool) -> String,
+    instance: fn() -> Option<&'static GameMgr>,
+}
```

Comment:
You don't need to define functions prototypes on a struct, you can just `impl` them, it doesn't affect the structs layout in memory either so you can put these functions in an `impl ZTGameMgr` block and it won't break `get_from_memory::<ZTGameMgr>`

Reply:
Oops. Rust learning pains.